spring-cloud-dataflow-platform-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/server/config/cloudfoundry/CloudFoundryTaskPlatformAutoConfiguration.java

```diff
@@ -38,6 +41,13 @@ public TaskPlatform cloudFoundryPlatform(CloudFoundryTaskPlatformFactory cloudFo
 		return cloudFoundryTaskPlatformFactory.createTaskPlatform();
 	}
 
+	@ConditionalOnCloudPlatform(CloudPlatform.CLOUD_FOUNDRY)
+	@Primary
+	@Bean
+	public TaskPlatform primaryTaskPlatform(CloudFoundryTaskPlatformFactory cloudFoundryTaskPlatformFactory) {
+		return cloudFoundryPlatform(cloudFoundryTaskPlatformFactory);
```

Comment:
Seems like we will have these beans created even if tasks are disabled.  I think we need to add `ConditionalOnTasksEnabled` to the class definition.

Reply:
Problem is that class is living in server-core... I wonder why we have this as an autoconfiguration then....

Reply:
Problem is that `ConditionalOnTasksEnabled` is in the core module.

Reply:
Plan was that all auto-config classes would eventually go to `spring-cloud-dataflow-autoconfigure` which can then access everything.